### PR TITLE
Set default value for port number and except parsing errors

### LINKED ISSUE #30: Skill Installation failure
## Expected Behavior

Installation of HomeAssistantSkill for Mycroft

## Steps to Reproduce the Problem

  1. Fresh build of Mycroft Mark1 image (```mark-1_production_20180301_build-18.zip```)
  2. Pairing of Mark1 to http://home.mycroft.ai
  3. Enable SSH on Mark1 via System Menu
  4. Connect via SSH to Mark1
  5. ```sudo su -```
  6. ```su - mycroft```
  7. Installation with ```msm install https://github.com/btotharye/mycroft-homeassistant.git```

#### ```mycroft-skills.log``` output:
```
12:42:48.191 - mycroft.skills.core:load_skill:106 - INFO - ATTEMPTING TO LOAD SKILL: mycroft-homeassistant with ID -745951192
12:42:50.459 - mycroft.skills.core:load_skill:143 - ERROR - Failed to load skill: mycroft-homeassistant
Traceback (most recent call last):
File "/usr/local/lib/python2.7/site-packages/mycroft_core-18.2.5-py2.7.egg/mycroft/skills/core.py", line 115, in load_skill
skill = skill_module.create_skill()
File "/opt/mycroft/skills/mycroft-homeassistant/__init__.py", line 592, in create_skill
return HomeAssistantSkill()
File "/opt/mycroft/skills/mycroft-homeassistant/__init__.py", line 166, in __init__
self._setup()
File "/opt/mycroft/skills/mycroft-homeassistant/__init__.py", line 178, in _setup
int(self.settings.get('portnum')),
ValueError: invalid literal for int() with base 10: ''
```

## Specifications
  - Version: ```mycroft-core 18.2.5```
  - Platform: ```Mycroft Mark1```

### PR DESCRIPTION
**Pull request by nielstron**
## Description:

**Related issue (if applicable):** fixes #30 

## Checklist:
  - [x] Travis is passing tests **Your PR cannot be merged unless tests pass**
  - [x] No PEP Bot Errors or Issues

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -170,12 +170,18 @@ def __init__(self):
...
+            try:
+                portnumber = int(portnumber)
+            except ValueError:
+                # String might be some rubbish (like '')
+                portnumber = 0
             self.ha = HomeAssistantClient(
                 self.settings.get('host'),
                 self.settings.get('password'),
-                int(self.settings.get('portnum')),
+                portnumber,
```
Comment (merps) [inline __init__.py]: Silly question - self reference removed why?

File: settingsmeta.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
                       "name": "portnum",
                       "type": "Number",
                       "label": "Port number",
-                      "value": ""
+                      "value": "8123"
```
Comment (merps) [inline settingsmeta.json]: This doesn't appear to load default value?

File: __init__.py (hunk 1 above)
Comment (nielstron) [reply __init__.py]: defined port number as variable 3-4 lines above

File: settingsmeta.json (hunk 2 above)
Comment (nielstron) [reply settingsmeta.json]: The values in settingsmeta.json are default values for the settings seen in home.mycroft.ai and self.settings, are overridden by user afterwards.

**Pull request merged**